indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
+            # break
+            elif self._ttPDFFormat.showBreaksAtConfLevel() and entry.type == TimetableEntryType.BREAK:
+                break_entry = entry
+                break_ = break_entry.object
+                room = ''
+                if break_.location_data['room_name']:
+                    room = ' - {}'.format(escape(break_.location_data['room_name']))
+
+                text = '<u>{}</u>{} ({}-{})'.format(escape(break_.title), room,
+                                                    format_time(break_entry.start_dt, timezone=self._tz),
+                                                    format_time(break_entry.end_dt, timezone=self._tz))
+
+                p1 = Paragraph(text, self._styles["session_title"])
                 res.append(p1)
+
                 if self._ttPDFFormat.showTitleSessionTOC():
-                    self._indexedFlowable[p1]={"text":escape(breakE.getTitle()), "level":2}
-                if entry == entriesOnDay[-1]: # if it is the last one, we do the page break and remove the previous one.
-                    i=len(res)-1
-                    while i>=0:
+                    self._indexedFlowable[p1] = {'text': escape(break_.title), 'level': 2}
+
+                if entry == entries[-1]:  # if it is the last one, we do the page break and remove the previous one.
+                    i = len(res) - 1
+                    while i >= 0:
                         if isinstance(res[i], PageBreak):
                             del res[i]
                             break
-                        i-=1
+                        i -= 1
                     if self._ttPDFFormat.showNewPagePerSession():
                         res.append(PageBreak())
-
         return res
 
-
-    def getBody(self,story=None):
+    def getBody(self, story=None):
         self._defineStyles()
         if not story:
-            story=self._story
+            story = self._story
         if not self._ttPDFFormat.showCoverPage():
             s = ParagraphStyle({})
             s.fontName = "Times-Bold"
             s.fontSize = 18
             s.leading = 22
             s.alignment = TA_CENTER
-            p=Paragraph(escape(self._conf.getTitle()),s)
+            p = Paragraph(escape(self._conf.getTitle()), s)
             story.append(p)
-            story.append(Spacer(1,0.4*inch))
+            story.append(Spacer(1, 0.4 * inch))
 
-        currentDay=self._conf.getSchedule().getAdjustedStartDate(self._tz)
-        while currentDay.strftime("%Y-%m-%d")<=self._conf.getSchedule().getAdjustedEndDate(self._tz).strftime("%Y-%m-%d"):
-            if len(self._showDays)>0 and \
-                    currentDay.strftime("%d-%B-%Y") not in self._showDays:
-                currentDay+=timedelta(days=1)
+        current_day = self._conf.getSchedule().getAdjustedStartDate(self._tz)
+        while (current_day.strftime("%Y-%m-%d") <=
+                self._conf.getSchedule().getAdjustedEndDate(self._tz).strftime("%Y-%m-%d")):
```

Comment:
Holy mother of god. This is just awful. Even if it's legacy code, it costs nothing to use `.date()` comparison.